components/match2/commons/match_group_input.lua

```diff
@@ -153,27 +167,41 @@ function MatchGroupInput._fetchBracketDatas(templateId, bracketId)
 		return (bracketId or '') .. '_' .. baseMatchId
 	end
 
+	-- Convert 0 based array to 1 based array
+	local function shiftArrayIndex(elems)
+		return Array.extend(elems[0], elems)
+	end
+
 	return Table.map(matches, function(_, match)
 		local bracketData = match.match2bracketdata
-		if String.nilIfEmpty(bracketData.toupper) then
-			bracketData.toupper = replaceBracketId(bracketData.toupper)
-		end
-		if String.nilIfEmpty(bracketData.tolower) then
-			bracketData.tolower = replaceBracketId(bracketData.tolower)
-		end
-		if String.nilIfEmpty(bracketData.thirdplace) then
-			bracketData.thirdplace = replaceBracketId(bracketData.thirdplace)
-		end
-		if String.nilIfEmpty(bracketData.bracketreset) then
-			bracketData.bracketreset = replaceBracketId(bracketData.bracketreset)
-		end
+
+		-- Convert 0 based array to 1 based array
+		bracketData.loweredges = bracketData.loweredges and shiftArrayIndex(bracketData.loweredges)
+		bracketData.lowerMatchIds = bracketData.lowerMatchIds and shiftArrayIndex(bracketData.lowerMatchIds)
```

Comment:
Why are they 0-based arrays in the first place?

Reply:
the bracket data is fetched from php which is zero based

this conversion can be done away with once we have cross wiki lpdb support

Reply:
OK

Reply:
Can we have a TODO comment on that, what needs to be done for dropping the array shift?

Reply:
Why can the conversion be dropped with cross wiki LPDB support? It's a purely theoretical construct at this point, we have no guarantees at all whether it will work nor how it will work.

Reply:
If cross wiki lpdb calls work similarly to current lpdb calls then it retrieves bracketdata as a json string. Parsing the json string in lua results in 1-indexed arrays.

I don't think there are other benefits besides less conversion. So not a strong reason to convert to cross wiki lpdb calls over what we have now.